src/allmydata/web/directory.py

```diff
@@ -63,6 +63,9 @@ def __init__(self, client, node, parentnode=None, name=None):
         self.parentnode = parentnode
         self.name = name
 
+        # probably better to just pass this in? can we?
+        self._operations = client.getServiceNamed("webish").getServiceNamed("operations")
```

Comment:
I don't see why not.

Reply:
Passing in "webish" alongside "client" has a few ripple effects .. and is using `getServiceNamed` a bad thing anyway?

Reply:
The only bad thing is that the string "webish" is a really weak link between this consumer and the provider (wherever that is).  A stronger link would make this better, I think.

For example, `client` could have a `getWebService` method that does exactly the same thing.  I would say that `client.getWebService()` and `client.getServiceNamed("webish")` introduce exactly the same amount of coupling between the two pieces of code but `client.getWebService()` makes the coupling more straightforward than `client.getServiceNamed("webish")`.

Mainly what I'm thinking about is that if someone changes how the "webish" service in client works, they have a clear responsibility to make sure `getWebService` keeps working (or they have an easily greppable term to go find and fix all of the consumers) because it's right there in the unit they're working on.

Reply:
Okay, I can easily make that change (that is "getters" that are more-explicitly tied / named). I did look at "what about passing in `webish`" which I think will take too much re-factoring for this already-big PR...

Reply:
Hmm, I guess one other point for this is that the "service" hierarchies have "refs to parents" .. and adding these getters would introduce "refs to children" so I guess "hopefully the garbage collector works" (but also: the `_Client` instance lasts approximately as long as the process).